src/lib.rs

```diff
@@ -634,16 +662,24 @@ pub fn unfill(text: &str) -> (String, Options<'_>) {
     }
 
     let mut unfilled = String::with_capacity(text.len());
-    for (idx, line) in trimmed.split('\n').enumerate() {
-        if idx == 0 {
+    let mut detected_line_ending = None;
+
+    for (line, ending) in line_ending::NonEmptyLines(text) {
+        if unfilled.is_empty() {
             unfilled.push_str(&line[options.initial_indent.len()..]);
         } else {
             unfilled.push(' ');
             unfilled.push_str(&line[options.subsequent_indent.len()..]);
         }
+        match (detected_line_ending, ending) {
+            (None, Some(_)) => detected_line_ending = ending,
+            (Some(LineEnding::CRLF), Some(LineEnding::LF)) => detected_line_ending = ending,
```

Comment:
In case we have mixed line terminators I decided to give `LF` a higher priority. So once we've encountered `LF`, we return it in `Options`

My motivation is that `LF` is more "universal" than `CRLF`: splitting a mixed-terminator file with `CRLF` will fail to split some lines, while splitting with `LF` will not miss lines but will leave some garbage `CR`s. That said, `unfill` still handles both `LF` and `CRLF` with no garbage `CR`'s in case of mixed input.

Reply:
Will do. As I said, I might not be able to finish it this week, but ultimately I intend to ship this with tests and documentation 

- [ ] mention `LineEnding::LF` is default in case of mixed input in the docs